crates/router/src/connector/cryptopay/transformers.rs

```diff
@@ -0,0 +1,240 @@
+use reqwest::Url;
+use serde::{Deserialize, Serialize};
+
+use crate::{
+    core::errors,
+    services,
+    types::{self, api, storage::enums},
+};
+
+//TODO: Fill the struct with respective fields
+#[derive(Default, Debug, Serialize, Eq, PartialEq)]
+pub struct CryptopayPaymentsRequest {
+    price_amount: i64,
+    price_currency: enums::Currency,
+    pay_currency: String,
+    success_redirect_url: Option<String>,
+    unsuccess_redirect_url: Option<String>,
+}
+
+impl TryFrom<&types::PaymentsAuthorizeRouterData> for CryptopayPaymentsRequest {
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(item: &types::PaymentsAuthorizeRouterData) -> Result<Self, Self::Error> {
+        let pay_currency = match item.request.payment_method_data {
+            api::PaymentMethodData::Crypto(ref cryptodata) => {
+                let pay_currency = cryptodata.pay_currency.clone().ok_or(
+                    errors::ConnectorError::MissingRequiredField {
+                        field_name: "pay_currency",
+                    },
+                )?;
```

Comment:
add `get_pay_currency` in connector/utils.rs
```suggestion
                let pay_currency = cryptodata.get_pay_currency()?;
```

Reply:
sure, done